bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueBridgeHandler.java

```diff
@@ -178,44 +173,55 @@ private boolean isReachable(String ipAddress) {
         @Override
         protected void doConnectedRun() throws IOException, ApiException {
             Map<String, FullSensor> lastSensorStateCopy = new HashMap<>(lastSensorStates);
+            lastSensorStates.clear();
+
+            final SensorStatusListener discovery = discoveryListener;
```

Comment:
Please keep the different states (added, removed and changed) for each kind of listener, without assuming what methods will be used by each listener.

Reply:
I have done a refactoring of discovery service. Please have a look.